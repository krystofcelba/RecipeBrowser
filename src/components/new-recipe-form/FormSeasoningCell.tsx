import React from 'react';

import FormInputCell from './FormInputCell';
import { NewSeasoning } from '../../redux/reducers/ui';
import i18n from '../../assets/i18n';

interface Props {
  seasoning: NewSeasoning;
  onChange: (seasoning: NewSeasoning) => void;
}

const FormSeasoningCell = ({ seasoning, onChange }: Props) => (
  <FormInputCell
    placeholder={i18n.t('seasoning')}
    value={seasoning.seasoning}
    onChangeText={text => onChange({ ...seasoning, seasoning: text })}
  />
);

export default FormSeasoningCell;
