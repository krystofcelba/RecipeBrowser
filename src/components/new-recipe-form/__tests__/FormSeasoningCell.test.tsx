import React from 'react';
import { shallow } from 'enzyme';

import FormSeasoningCell from '../FormSeasoningCell';

describe('FormSeasoningCell component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<FormSeasoningCell seasoning={{ id: 1, seasoning: 'test' }} onChange={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('call onChange as expected', () => {
    const onChangeSpy = jest.fn();
    const wrapper = shallow(<FormSeasoningCell seasoning={{ id: 1, seasoning: 'test' }} onChange={onChangeSpy} />);
    wrapper.simulate('changeText', 'a');
    expect(onChangeSpy.mock.calls).toMatchSnapshot();
  });
});
