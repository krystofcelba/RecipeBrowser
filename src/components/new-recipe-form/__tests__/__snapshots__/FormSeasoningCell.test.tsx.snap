// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSeasoningCell component call onChange as expected 1`] = `
Array [
  Array [
    Object {
      "id": 1,
      "seasoning": "a",
    },
  ],
]
`;

exports[`FormSeasoningCell component renders as expected 1`] = `
<FormInputCell
  onChangeText={[Function]}
  placeholder="Seasoning"
  value="test"
/>
`;
